addons/mail/static/src/core/messaging_service.js

```diff
@@ -266,7 +270,12 @@ export class Messaging {
                     break;
                 case "mail.message/inbox": {
                     const data = Object.assign(notif.payload, { body: markup(notif.payload.body) });
-                    this.messageService.insert(data);
+                    const message = this.messageService.insert(data);
+                    const inbox = this.store.discuss.inbox;
+                    if (!inbox.messages.includes(message)) {
```

Comment:
comparing objects is dangerous with proxy

Reply:
Cost nothing to change those `includes` by `some(({ id }) => id === message.id)`. Updated :+1:

Reply:
> Cost nothing to change those `includes` by `some(({ id }) => id === message.id)`. Updated 👍

It does cost readability, which is more important than to fix an issue that doesn't exist